duckduckgo/iOS
# privacy protection whitelisting

Reviewer: Mia
Asana: https://app.asana.com/0/414709148257752/455982666371468
CC:

**Description**:

* Allow user to add/remove current site to the whitelist via the action menu
* Show the whitelist in settings (position is subject to change based on future work)
* Add/remove from whitelist via settings
* Bonus: Allow user to add/remove current site from bookmarks via the action menu
* Bonus: Fix memory leak around fire button click handler

Note that some of the colours of the navigation bar buttons are blue but in the designs are white.  This is consistent with bookmarks, so will get some clarity around making this consistent across the app separately.

**Steps to test this PR**:

Test 1:
1. Visit a site
1. Add it to the whitelist via the menu.  The site will reload
1. Visit settings the, the site will be listed
1. Go back to the tab and remove it from the whitelist via the menu. The site will reload

Test 2:
1. Open the whitelist from settings
1. Tap add
1. Enter something that isn't a domain and click Add
1. The domain is not added

Test 3
1. Open the whitelist from settings
1. Tap add
1. Enter a valid domain and click Cancel
1. The domain is not added

Test 4:
1. Open the whitelist from settings
1. Tap add
1. Enter a valid domain and click Add
1. The domain is displayed in the list
1. Visit the website in a new tab
1. Open the menu... the action says Remove from Whitelist
1. Return to the whitelist in settings
1. Swipe to delete the domain
1. Return to the tab for the website
1. Open the menu... the action says Add to Whitelist



###### Reviewer Checklist:


###### PR DRI Checklist:
